plugins/backstage-plugin-flux/src/hooks/useGetUser.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  IdentityApi,
+  identityApiRef,
+  useApi,
+} from '@backstage/core-plugin-api';
+import { UserIdentity } from '@backstage/core-components';
+import { useQuery } from '@tanstack/react-query';
+
+export async function getUserInfo(identityApi: IdentityApi) {
+  const backstageIdentity = await identityApi.getBackstageIdentity();
+  const profile = await identityApi.getProfileInfo();
+
+  return await UserIdentity.fromLegacy({
+    userId: backstageIdentity.userEntityRef,
+    profile: profile,
+  });
+}
+
+/**
+ *
+ * @public
+ */
+export function useGetUserInfo() {
+  const identityApi = useApi(identityApiRef);
```

Comment:
Righttt... we always need to use a hook to interact w/ the apis?

I was thinking we could move the user data fetching into some simple promise

```
const user = await getUserData();
toggleSuspend(..., user)
```

I suppose we still could, e.g. return `getUserInfo` from this hook.

Reply:
We could potentially though we need to make changes further up b/c the await requires the function to be async which results in hook rules errors etc. Tbh I like the pattern that we're using throughout and how the complexity around async remains isolated to the hook but if you want I can try to remove the hook